### PR TITLE
Allocation reduction

### PR DESCRIPTION
**Pull request by aconz2**
### Summary of the PR

I stumbled upon this code in profiling cloud hypervisor and seeing some vec things show up. I can't measure any real difference in my cloud hypervisor testing but there is a measurable difference in the tests. Not sure if you want it but figured I'd open a PR

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR have Signed-Off-By trailers (with
  `git commit -s`), and the commit message has max 60 characters for the
  summary and max 75 characters for each description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Comment by aconz2** [comment]
see the [commit message](https://github.com/rust-vmm/acpi_tables/pull/39/commits/66cb82b59668bf4cbae250882054e36e82b29de0) for some more details/numbers

File: src/lib.rs [hunk 1]
```diff
@@ -87,6 +81,10 @@ impl AmlSink for alloc::vec::Vec<u8> {
     fn byte(&mut self, byte: u8) {
         self.push(byte);
     }
+
+    fn vec(&mut self, v: &[u8]) {
```
Comment (rbradford) [inline src/lib.rs:85]: I wonder if we should rename this API to e.g.

`AmlSink::sink_{byte, word, dword, qword, slice}` @sboeuf @timw-rivos WDYT?

**Review by rbradford — COMMENTED** [review]
Looks good - thanks!

File: src/aml.rs [hunk 2]
```diff
@@ -1779,6 +1779,16 @@ mod tests {
         assert_eq!(aml, &mbrd_scope[..]);
     }
```
[the remark this answers is not in the document]
Comment (aconz2) [reply src/aml.rs:1742]: updated

File: src/aml.rs [hunk 3]
```diff
@@ -273,13 +269,10 @@ impl<'a> Aml for Package<'a> {
             child.to_aml_bytes(&mut bytes);
         }
 
-        let mut pkg_length = create_pkg_length(bytes.len(), true);
```
[the remark this answers is not in the document]
Comment (timw-rivos) [reply src/aml.rs:276]: I think I just thought it through once and didn't look at it again ;)

File: src/lib.rs (hunk 1 above)
Comment (timw-rivos) [reply src/lib.rs:85]: It is private to this library, but it is more clear

**Review by timw-rivos — DISMISSED** [review]

**Comment by aconz2** [comment]
was out last week, just fixed the cargo fmt errors

**Review by timw-rivos — APPROVED** [review]

**Review by rbradford — APPROVED** [review]
Thank you for your contribution and patience through the review process!

**Pull request merged**